src/vs/workbench/contrib/positronConsole/browser/positronConsoleActions.ts

```diff
@@ -566,6 +567,42 @@ export function registerPositronConsoleActions() {
 		}
 	});
 
+	/**
+	 * Register the "Execute without advancing" action. This action executes the
+	 * current statement in the editor without advancing the cursor.
+	 */
+	registerAction2(class extends Action2 {
+		constructor() {
+			super({
+				id: 'workbench.action.positronConsole.executeCodeWithoutMotion',
+				title: {
+					value: localize('workbench.action.positronConsole.executeCodeWithoutMotion', 'Execute Code Without Moving)'),
+					original: 'Execute Code Without Moving'
+				},
+				f1: false,
+				category: POSITRON_CONSOLE_ACTION_CATEGORY,
+				precondition: ContextKeyExpr.and(
+					EditorContextKeys.editorTextFocus,
+					NOTEBOOK_EDITOR_FOCUSED.toNegated()
+				),
+				keybinding: {
+					weight: KeybindingWeight.BuiltinExtension,
```

Comment:
```suggestion
					weight: KeybindingWeight.WorkbenchContrib,
```

For consistency with `PositronConsoleCommandId.ExecuteCode` (I think it will be pretty confusing if these are different)